script/migrate_from_ips_to_data_centers/copy_data_to_data_centers.rb

```diff
@@ -0,0 +1,139 @@
+end
+
+def create_or_update_validator_ip(ip, data_center_host)
+  val_ip = ValidatorIp.find_by(address: ip.address)
+
+  if val_ip
+    val_ip.update(
+      data_center_host_id: data_center_host.id,
+      traits_ip_address: ip.traits_ip_address,
+      traits_network: ip.traits_network,
+      traits_domain: ip.traits_domain
+    )
+    @file.write "Validator IP with id #{val_ip.id} updated with data center host id: #{val_ip.data_center_host_id} assigned to data center: #{data_center_host.data_center_key}.\n"
+  else
+    val_ip = ValidatorIp.create!(
+      address: ip.address,
+      data_center_host_id: data_center_host.id,
+      traits_ip_address: ip.traits_ip_address,
+      traits_network: ip.traits_network,
+      traits_domain: ip.traits_domain
+    )
+
+    @file.write "Validator IP created for address #{ip.address} with data center host id: #{val_ip.data_center_host_id} assigned to data center: #{data_center_host.data_center_key}.\n"
+  end
+end
+
+### ### Ip overrirdes  ### ###
+
+def update_validator_ip_from_ip_override(ip_override)
+  val_ip = ValidatorIp.find_by(address: ip_override.address)
+  ip = Ip.find_by(address: ip_override.address)
+
+  data_center = DataCenter.find_by(
+    traits_autonomous_system_number: ip_override.traits_autonomous_system_number,
+    traits_autonomous_system_organization: ip_override.traits_autonomous_system_organization,
+    country_iso_code: ip_override.country_iso_code,
+    country_name: ip_override.country_name,
+    data_center_key: ip_override.data_center_key
+  )
+
+  data_center_host = DataCenterHost.find_or_create_by!(
+    host: ip_override.data_center_host,
+    data_center: data_center
+  )
+
+  if data_center_host && val_ip
+    val_ip.update(is_overridden: true, data_center_host_id: data_center_host.id)
+
+    @file.write "Validator IP with id: #{val_ip.id} 
+                updated with data center host  #{ip_override.data_center_host}, id: #{data_center_host.id},
+                ip_override.data_center_key: #{ip_override.data_center_key}, data_center.data_center_key: #{data_center.data_center_key}
+                is_overridden set to true.\n"
+  else
+    @file.write "Data center host not found for
+                  host: #{ip_override.data_center_host}.
+                  ip_override: #{ip_override.id}.\n" unless data_center_host
+    # @file.write "Data center host not found for
+    #               traits_autonomous_system_number: #{ip_override.traits_autonomous_system_number}, 
+    #               traits_autonomous_system_organization: #{ip_override.traits_autonomous_system_organization}, 
+    #               data_center_key: #{ip_override.data_center_key},
+    #               data_center_host: #{ip_override.data_center_host}.
+    #               ip_override: #{ip_override.id}.\n" unless data_center
+    @file.write "Validator IP not found for id: #{ip_override.id}, ip address #{ip_override.address}.\n" unless ip_override
+  end
+end
+
+Ip.find_in_batches do |batch|
```

Comment:
I would find only IP's that has no corresponding ValidatorIp, just in case we have to run this twice or something brakes.